test/e2e/artifacts_test.go

```diff
@@ -391,6 +407,8 @@ func (s *ArtifactsSuite) TestArtifactGC() {
 		when.
 			DeleteWorkflow().
 			WaitForWorkflowDeletion()
```

Comment:
```suggestion
			WaitForWorkflowDeletion().
```

Lint is [failing](https://github.com/argoproj/argo-workflows/actions/runs/9455186352/job/26044675796?pr=13066) as the previous line needs a trailing dot now for the chain